testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveViewsCompatibility.java

```diff
@@ -0,0 +1,140 @@
+            onTrino().executeQuery("CREATE VIEW iceberg.default.iceberg_view_qualified_iceberg AS SELECT * FROM iceberg.default.iceberg_table");
+            onTrino().executeQuery("CREATE VIEW iceberg.default.iceberg_view_unqualified_iceberg AS SELECT * FROM iceberg_table");
+
+            // select some random catalog so we are not biased towards iceberg.default during assertions
+            onTrino().executeQuery("USE tpch.tiny");
+
+            // both hive and iceberg catalogs should list all the views.
+            assertThat(onTrino().executeQuery("SHOW TABLES FROM hive.default"))
+                    .containsOnly(ImmutableList.<QueryAssert.Row>builder()
+                            .addAll(hivePreexistingTables)
+                            .add(row("hive_table"))
+                            .add(row("iceberg_table")) // TODO; should this be filtered out?
+                            .add(row("hive_view_qualified_hive"))
+                            .add(row("hive_view_unqualified_hive"))
+                            .add(row("hive_view_qualified_iceberg"))
+                            .add(row("hive_view_unqualified_iceberg"))
+                            .add(row("iceberg_view_qualified_hive"))
+                            .add(row("iceberg_view_qualified_iceberg"))
+                            .add(row("iceberg_view_unqualified_iceberg"))
+                            .build());
+
+            assertThat(onTrino().executeQuery("SHOW TABLES FROM iceberg.default"))
+                    .containsOnly(ImmutableList.<QueryAssert.Row>builder()
+                            .addAll(icebergPreexistingTables)
+                            .add(row("iceberg_table"))
+                            .add(row("hive_view_qualified_hive"))
+                            .add(row("hive_view_unqualified_hive"))
+                            .add(row("hive_view_qualified_iceberg"))
+                            .add(row("hive_view_unqualified_iceberg"))
+                            .add(row("iceberg_view_qualified_hive"))
+                            .add(row("iceberg_view_qualified_iceberg"))
+                            .add(row("iceberg_view_unqualified_iceberg"))
+                            .build());
+
+            // try to access all views via hive catalog
+            assertThat(onTrino().executeQuery("SELECT * FROM hive.default.hive_view_qualified_hive")).containsOnly(row(1));
+            assertThat(onTrino().executeQuery("SELECT * FROM hive.default.hive_view_unqualified_hive")).containsOnly(row(1));
+            assertThat(onTrino().executeQuery("SELECT * FROM hive.default.hive_view_qualified_iceberg")).containsOnly(row(2));
+            assertThatThrownBy(() -> onTrino().executeQuery("SELECT * FROM hive.default.hive_view_unqualified_iceberg"))
+                    // hive connector tries to read from iceberg table
+                    // TODO; make query fail with nicer message
```

Comment:
nit: are `;` instead of `:` intentional? it's not recognized by intellij

Reply:
thanks!